### PR TITLE
[backport v1.0] fix: fix intermediate certificate validaiton

### PR DESCRIPTION
**Pull request by asraa**
Fixes https://github.com/slsa-framework/slsa-github-generator/issues/762

### CODE REVIEW

File: pkg/provenance.go [hunk 1]
```diff
@@ -186,7 +187,19 @@ func verifyTlogEntry(ctx context.Context, rekorClient *client.Rekor, uuid string
...
 	}
-	e := lep.Payload[params.EntryUUID]
+
+	uuid, err := sharding.GetUUIDFromIDString(params.EntryUUID)
+	if err != nil {
+		return nil, err
+	}
+
+	var e models.LogEntryAnon
+	for k, entry := range lep.Payload {
```
Comment (laurentsimon) [inline pkg/provenance.go:197]: so we're expecting a single UUID?

**Review by laurentsimon — APPROVED** [review]

File: pkg/provenance.go (hunk 1 above)
Comment (asraa) [reply pkg/provenance.go:197]: Yes, this entry point retrieves exactly one endpoint. but let me check this. i did not put the correct merge into branch in the PR.

that commit was already in 1.0.1

Comment (asraa) [reply pkg/provenance.go]: fixed, sorry about that. it was trying to merge asraa:release/v1.0 commits into main, not into upstream:release/v1.0

Comment (laurentsimon) [reply pkg/provenance.go]: np.

**Comment by asraa** [comment]
I will trigger the 1.0.3 release after merge

**Pull request merged**